### PR TITLE
server: use single flight to load global variables

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?

Problem Summary:
Burst in connection build may lead to high meta region coprocessor CPU usage

### What is changed and how it works?

What's Changed:
Use single flight to load global variables, execute only one select statement to fetch the global variables from store, and update the cache only once for the same results

How it Works:

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
Do some tests build connection and then close without connection pooling, 
the master branch may raise about 
```
grep "for debug" tidblog  | wc -l
683
```
time select global variables queries during tests

the dev branch will
```
grep "for debug" tidblog | wc -l
6
```

Side effects

### CODE REVIEW

File: domain/global_vars_cache.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/golang/groupcache/singleflight"
```
Comment (coocood) [inline domain/global_vars_cache.go:21]: Have you compared this with `golang.org/x/sync/singleflight` ?

Comment (cfzjywxk) [reply domain/global_vars_cache.go:21]: [golang.org/x/sync/singleflight](https://github.com/golang/sync/blob/master/singleflight/singleflight.go) is a later version, which has more interfaces and fields than [golang/groupcache/singleflight](https://github.com/golang/groupcache/blob/master/singleflight/singleflight.go), better to use the sync package one

**Comment by coocood** [comment]
LGTM

File: domain/global_vars_cache.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ package domain
 
 import (
 	"fmt"
+	"golang.org/x/sync/singleflight"
```
Comment (lysu) [inline domain/global_vars_cache.go:18]: need fmt pkg :D

reset LGTM

**Pull request merged**